Cargo.lock

```diff
@@ -106,12 +106,14 @@ dependencies = [
  "convert 0.3.0",
  "domain 0.2.3 (registry+https://github.com/rust-lang/crates.io-index)",
  "env_logger 0.5.3 (registry+https://github.com/rust-lang/crates.io-index)",
+ "flate2 1.0.1 (registry+https://github.com/rust-lang/crates.io-index)",
  "futures 0.1.18 (registry+https://github.com/rust-lang/crates.io-index)",
  "futures-mpsc-lossy 0.3.0",
  "h2 0.1.3 (registry+https://github.com/rust-lang/crates.io-index)",
  "http 0.1.5 (registry+https://github.com/rust-lang/crates.io-index)",
  "httparse 1.2.4 (registry+https://github.com/rust-lang/crates.io-index)",
  "hyper 0.11.22 (registry+https://github.com/rust-lang/crates.io-index)",
+ "hyper-compress 0.0.1 (git+https://github.com/hawkw/hyper-compress.git)",
```

Comment:
I'm not convinced we need a generalized gzip hyper middleware. We just need to, maybe, compress 1 thing, telemetry reports. I suspect the amount of code to do that compression directly in conduit-proxy's telemetry code is probably smaller than the number of lines added to Conduit's Cargo.lock in this PR. I don't anticipate that we're going to start compressing/decompressing other kinds of request/response bodies soon in conduit-proxy.